File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.py
import numpy as np
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        c=np.base_repr(n,base=2)
        print(c)
        d=""
        for i in c:
            if i == "1":d+='0'
            elif i == '0':d+='1'
        return int(d,2)

File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.cpp
class Solution {
public:
    string toBit(int n){
        string s = "";
        while(n > 0){
            s = char((n % 2) + '0') + s;
            n /= 2;
        }
        return s;
    }
    int toInt(string bit){
        reverse(bit.begin(), bit.end());
        int sum = 0;
        for(int i=0;i<bit.size();i++){
            if(bit[i] == '1') sum += pow(2, i);
        }
        return sum;
    }
    int bitwiseComplement(int n) {
        if(n == 0) return 1;
        string bit = toBit(n);
        for(int i=0;i<bit.size();i++){
            if(bit[i] == '0'){
                bit[i] = '1';
            }else{
                bit[i] = '0';
            }
        }
        cout << bit;
        return toInt(bit);
    }
};